docs/source/API/core/memory_spaces.rst

```diff
@@ -152,6 +153,26 @@ Typedefs
 * ``memory_space``: The self type;
 * ``execution_space``: the default |ExecutionSpace|_ to use when constructing objects in memory provided by an instance of ``MemorySpace``, or (potentially) when deep copying from or to such memory (see |DeepCopyDocumentation|_ for details). Kokkos guarantees that ``Kokkos::SpaceAccessibility<execution_space, memory_space>::accessible`` will be ``true`` (see |KokkosSpaceAccessibility|_).
 * ``device_type``: ``DeviceType<execution_space,memory_space>``.
+* ``size_type``: the default integer type associated with this space. Used as preferred type for indexing.
+
+.. list-table::
+ :widths: 25 25
+ :header-rows: 1
+
+ * - Space
+   - ``size_type``
+ * - ``Cuda*``
+   - ``unsigned int``
+ * - ``HIP*``
+   - ``unsigned int``
+ * - ``SYCL*``
+   - ``int``
```

Comment:
```suggestion
 * - ``Cuda``
   - ``unsigned int``
 * - ``HIP``
   - ``unsigned int``
 * - ``SYCL``
   - ``int``
```
or am I missing why you are adding `*` to these three backends?

Reply:
hmm it was due to the type being linked to the `MemorySpace`, thus I added the wildcard.
But I agree, that it feels like it should be backend-specific (so far it is, but each `MemorySpace` impl could define it differently). We could decide to just communicate it as backend-specific, which we fulfill at the moment.

Reply:
It's not quite clear to me why a memory space needs to have a `size_type` (as opposed to the execution space)` and I would rather list that member there or at least additionally.

Reply:
changed to point to the execution space now